.circleci/config.yml

```diff
@@ -97,37 +91,73 @@ jobs:
       - verify-version-in-docker-image:
           image: montecarlodata/<< parameters.docker_hub_repository >>:latest-azure
           version: << parameters.code_version >>,<< pipeline.number >>
       - docker/push:
           image: montecarlodata/<< parameters.docker_hub_repository >>
-          tag: latest-generic,<< parameters.code_version >>-generic,latest-cloudrun,<< parameters.code_version >>-cloudrun,latest-lambda,<< parameters.code_version >>-lambda,latest-azure,<< parameters.code_version >>-azure
+          tag: latest-generic,<< parameters.code_version >>-generic,latest-cloudrun,<< parameters.code_version >>-cloudrun,latest-azure,<< parameters.code_version >>-azure
 
   build-and-push-lambda:
     machine:
-      image: ubuntu-2004:202201-02
+      image: ubuntu-2204:current
     parameters:
+      docker_hub_repository:
+        type: string
       aws_ecr_repository:
         type: string
       code_version:
         type: string
     steps:
       - checkout
-      - aws-ecr/build-and-push-image:
-          checkout: false
-          repo: << parameters.aws_ecr_repository >>
-          tag: << parameters.code_version >>
-          extra-build-args: --target lambda --build-arg code_version=<< parameters.code_version >> --build-arg build_number=<< pipeline.number >>
+      - docker/check:
+          use-docker-credentials-store: true
+      - docker/build:
+          step-name: Build lambda image
+          use-buildkit: true # to build only the required stages
+          extra_build_args: --target lambda --build-arg code_version=<< parameters.code_version >> --build-arg build_number=<< pipeline.number >>
+          image: montecarlodata/<< parameters.docker_hub_repository >>
+          tag: latest-lambda,<< parameters.code_version >>-lambda
+      - run:
+          name: Tag lambda image
+          command: |
+            docker tag montecarlodata/<< parameters.docker_hub_repository >>:<< parameters.code_version >>-lambda ${AWS_ECR_ACCOUNT_URL}/<< parameters.aws_ecr_repository >>:<< parameters.code_version >>
+            docker tag montecarlodata/<< parameters.docker_hub_repository >>:latest-lambda ${AWS_ECR_ACCOUNT_URL}/<< parameters.aws_ecr_repository >>:latest
       - verify-version-in-docker-image:
           image: ${AWS_ECR_ACCOUNT_URL}/<< parameters.aws_ecr_repository >>:<< parameters.code_version >>
           version: << parameters.code_version >>,<< pipeline.number >>
-      - aws-ecr/tag-image:
+      - docker/push:
+          image: montecarlodata/<< parameters.docker_hub_repository >>
+          tag: latest-lambda,<< parameters.code_version >>-lambda
+      - aws-cli/setup:
+          role_arn: "${AWS_ROLE_ARN}"
+      - aws-ecr/ecr_login
+      - aws-ecr/push_image:
           repo: << parameters.aws_ecr_repository >>
-          source-tag: << parameters.code_version >>
+          tag: latest,<< parameters.code_version >>
+  push-lambda-legacy-repo:
```

Comment:
Just so we don't forget cleaning this up, I created a time-bound ticket:  https://linear.app/montecarlodata/issue/KIL-3628/remove-legacy-agent-repo